src/logger_handler/logger_handler.go

```diff
@@ -54,42 +53,35 @@ func (handler *LoggerHandler) Log(loggable Loggable) {
 }
 
 func (handler *LoggerHandler) UpdateMessage(topic string, payload json.RawMessage, source string) {
-	handler.trace.Debug().Str("topic", topic).Str("source", source).Msg("update message")
+	handler.trace.Info().Str("topic", topic).Str("source", source).Msg("update message")
 	switch topic {
 	case handler.config.Topics.Enable:
-		handler.handleEnable(payload, source)
-	}
-
-	handler.notifyState()
-}
+		var enable bool
+		err := json.Unmarshal(payload, &enable)
+		if err != nil {
+			handler.trace.Error().Stack().Err(err).Msg("unmarshal enable")
+		}
 
-func (handler *LoggerHandler) handleEnable(payload json.RawMessage, source string) error {
-	var enable bool
-	err := json.Unmarshal(payload, &enable)
-	if err != nil {
-		handler.trace.Error().Stack().Err(err).Msg("unmarshal enable")
-		return err
+		handler.handleEnable(enable, source)
 	}
 
-	handler.updateState(enable, source)
-	return nil
+	handler.notifyState()
 }
 
-func (handler *LoggerHandler) updateState(enable bool, source string) error {
-	handler.trace.Debug().Bool("enable", enable).Str("source", source).Msg("update state")
-
+func (handler *LoggerHandler) handleEnable(enable bool, source string) error {
 	if !handler.verifySession(source) {
 		handler.trace.Warn().Str("source", source).Msg("tried to change running log session")
 		return fmt.Errorf("%s tried to change running log session of %s", source, handler.currentSession)
 	}
 
 	handler.isRunningMx.Lock()
 	if enable && !handler.isRunning {
+		handler.isRunningMx.Unlock()
 		handler.start()
 	} else if !enable && handler.isRunning {
+		handler.isRunningMx.Unlock()
 		handler.stop()
 	}
```

Comment:
put this in a separate function to defer the mutex unlock, it's more clear and less error prone
```suggestion
func (handler *LoggerHandler) changeState(enable bool) {
	handler.isRunningMx.Lock()
	defer handler.isRunningMx.Unlock()
	if enable && !handler.isRunning {
		handler.start()
	} else if !enable && handler.isRunning {
		handler.stop()
	}
}
```